lib/event-highlighter.js

```diff
@@ -0,0 +1,318 @@
+'use babel'
+
+import { Range, Point, CompositeDisposable } from "atom"; // eslint-disable-line no-unused-vars
+import path from "path"; // eslint-disable-line no-unused-vars
+const osc = require('osc-min');
+import OscServer from './osc-server';
+
+const CLASS = Object.freeze({
+  base: "event-highlight",
+  oscPrefix: "event-highlight-osc-",
+  idPrefix: "event-highlight-",
+});
+
+/** Helper that creates (or returns existing) nested maps. */
+function ensureNestedMap(root, key) {
+  if (!root.has(key)) root.set(key, new Map());
+  return root.get(key);
+}
+
+/**
+ * Responsible for receiving OSC events describing source‑code positions and
+ * visually highlighting those regions in Pulsar TextEditor.
+ */
+export default class EventHighlighter {
+
+  constructor(consoleView, editors) {
+    this.consoleView = consoleView;
+
+    this.editors = editors;
+
+    // Data‑structures -------------------------------------------------------
+    this.markers = new Map();              // textbuffer.id → row → col → Marker
+    this.highlights = new Map();           // eventId → texteditor.id → col → Marker
+    this.filteredMessages = new Map();     // eventId → event
+    this.receivedThisFrame = new Map();    // eventId → event
+    this.addedThisFrame = new Set();       // Set<event>
+    this.oscStyleIds = new Set();          // received ids
+    this.eventIds = [];                    // [{textbufferId, row}]
+
+    // Animation state -------------------------------------------------------
+    this.then = 0; // time at previous frame
+
+    // Bind instance methods used as callbacks -----------------------------
+    this.animate = this.animate.bind(this);
+  }
+
+  /** Initialise OSC listeners & start animation loop */
+  init() {
+    this.#installBaseHighlightStyle();
+
+    // Kick‑off animation loop
+    this.then = window.performance.now();
+    requestAnimationFrame(this.animate);
+  }
+
+  /** Clean‑up resources when package is deactivated */
+  destroy() {
+    try {
+      this.server?.destroy();
+    } catch (e) {
+      this.consoleView.logStderr(`OSC server encountered an error while destroying:\n${e}`);
+    }
+  }
+
+  // ----------------------------------------------------------------------
+  // Public API
+  // ----------------------------------------------------------------------
```

Comment:
this kind of comments are quite misleading, because there's some public api above (init, destroy), but also below the `Private Helpers` block, I'd just remove these blocks and put the public methods on top and the private on bottom.

Reply:
Done